packages/puppeteer-extra-plugin-stealth/evasions/user-agent-override/index.test.js

```diff
@@ -120,22 +120,42 @@ test('stealth: navigator.languages with custom locale', async t => {
   t.deepEqual(lang, 'de-DE')
 })
 
-test('stealth: navigator.platform with default platform', async t => {
-  const puppeteer = addExtra(vanillaPuppeteer).use(Plugin())
-  const browser = await puppeteer.launch({ headless: true })
-  const page = await browser.newPage()
+test('stealth: test if UA hints are correctly set', async t => {
+  const puppeteer = addExtra(vanillaPuppeteer).use(
+    Plugin({
+      userAgent:
+        'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.96 Safari/537.36'
+    })
+  )
 
-  const platform = await page.evaluate(() => navigator.platform)
-  t.true(platform === 'Win32')
-})
+  const browser = await puppeteer.launch({
+    headless: false, // only works on headful
+    args: ['--enable-features=UserAgentClientHint']
+  })
 
-test('stealth: navigator.platform with custom platform', async t => {
-  const puppeteer = addExtra(vanillaPuppeteer).use(
-    Plugin({ platform: 'MyFunkyPlatform' })
+  const majorVersion = parseInt(
+    (await browser.version()).match(/\/([^\.]+)/)[1]
   )
-  const browser = await puppeteer.launch({ headless: true })
+  if (majorVersion < 88) {
+    return t.true(true) // Skip test on browsers that don't support UA hints
+  }
+
   const page = await browser.newPage()
 
-  const platform = await page.evaluate(() => navigator.platform)
-  t.true(platform === 'MyFunkyPlatform')
+  await page.goto('https://headers.cf/headers/?format=raw')
+  const firstLoad = await page.content()
+  t.true(
+    firstLoad.includes(
+      `sec-ch-ua: "Chromium";v="88", "Google Chrome";v="88", ";Not A Brand";v="99"`
```

Comment:
Is there a chance `88` will be outdated once we upgrade pptr and the test will fail? 🤔

Reply:
Actually no, the version is coming from this line: https://github.com/berstend/puppeteer-extra/pull/413/files#diff-a9da9f27e487fcdb0773c1ec2bbb6a040cbd643f81fb66ecdfdf48238b6cd5cdR127

We could use a fake `99.99.99.99` version if you prefer that though.

Reply:
Ah, never mind - didn't realize the data is fully defined by us :-)

Reply:
Added some more tests now, used a fake version as well to be sure